### PR TITLE
User roles

### PR DESCRIPTION
**Pull request by lefreire**

### CODE REVIEW

File: dbaas/drivers/mongodb.py [hunk 1]
```diff
@@ -271,7 +277,8 @@ def info(self):
     def create_user(self, credential, roles=["readWrite", "dbAdmin"]):
         with self.pymongo(database=credential.database) as mongo_database:
             mongo_database.add_user(
-                credential.user, password=credential.password, roles=roles)
+                credential.user, password=credential.password,
+                roles=self.roles[credential.privileges])# roles=roles)
```
Comment (otherpirate) [inline dbaas/drivers/mongodb.py]: Is the # roles=roles necessary?

File: dbaas/drivers/mongodb.py [hunk 2]
```diff
@@ -271,7 +277,8 @@ def info(self):
     def create_user(self, credential, roles=["readWrite", "dbAdmin"]):
```
Comment (otherpirate) [inline dbaas/drivers/mongodb.py]: Is the role parameter necessary?

File: dbaas/drivers/mysqldb.py [hunk 3]
```diff
@@ -235,7 +242,7 @@ def create_database(self, database):
         LOG.info("creating database %s" % database.name)
         self.__query("CREATE DATABASE %s" % database.name)
 
-    def create_user(self, credential, roles=["ALL PRIVILEGES"]):
+    def create_user(self, credential, role=["ALL PRIVILEGES"]):
```
Comment (otherpirate) [inline dbaas/drivers/mysqldb.py]: Is role parameter necessary yet?

File: dbaas/logical/views.py [hunk 4]
```diff
@@ -57,13 +58,16 @@ class CredentialView(CredentialBase):
...
+        privileges = request.POST.get("privileges", None)
+
         try:
             database = get_object_or_404(Database, pk=database_id)
 
             # check permission
             self.check_permission(request, "logical.add_credential", database)
 
-            credential = Credential.create_new_credential(username, database)
+            credential = Credential.create_new_credential(username, database,
```
Comment (otherpirate) [inline dbaas/logical/views.py]: PEP8

File: dbaas/logical/static/js/credential.js [hunk 5]
```diff
@@ -160,21 +160,31 @@
...
                 "<a href='#' class='save-new-credential btn btn-primary'>Save</a>" +
-                "</td></tr>");
+                "</td></tr></div>");
         });
 
         $(document).on("click.save-new-credential", ".save-new-credential", function(e) {
             var $insert_row = $(e.target).parent().parent(),
-                username = $("input", $insert_row).val();
+                username = $("#user").val();
+                privileges = $("#privileges").val()
```
Comment (elesbom) [inline dbaas/logical/static/js/credential.js]: var privileges is global. Do the same with username.

File: dbaas/logical/static/js/credential.js [hunk 6]
```diff
@@ -215,12 +225,13 @@
...
             * Create a new credential on server and put on page
             */
-            create: function(username, $row, callback) {
+            create: function(username, privileges, $row, callback) {
                 var self = this;
                 $.ajax({
                     "url": "/logical/credential/",
                     "type": "POST",
-                    "data": { "username": username, "database_id": get_database_id() },
+                    "data": { "username": username, "database_id":
```
Comment (elesbom) [inline dbaas/logical/static/js/credential.js]: fix blank lines

File: dbaas/logical/static/js/credential.js [hunk 7]
```diff
@@ -160,21 +160,31 @@
...
+
                 "<a href='#' class='save-new-credential btn btn-primary'>Save</a>" +
-                "</td></tr>");
+                "</td></tr></div>");
         });
 
         $(document).on("click.save-new-credential", ".save-new-credential", function(e) {
             var $insert_row = $(e.target).parent().parent(),
-                username = $("input", $insert_row).val();
+                username = $("#user").val();
```
Comment (elesbom) [inline dbaas/logical/static/js/credential.js]: Be careful using generic Ids like #user and #credentials. Use css classes when u can :D

File: dbaas/drivers/mongodb.py [hunk 8]
```diff
@@ -29,6 +29,12 @@ class MongoDB(BaseDriver):
 
     RESERVED_DATABASES_NAME = ['admin', 'config', 'local']
 
+    roles = {
```
Comment (elesbom) [inline dbaas/drivers/mongodb.py]: Constants must be uppercase

File: dbaas/drivers/mysqldb.py [hunk 9]
```diff
@@ -235,7 +242,7 @@ def create_database(self, database):
         LOG.info("creating database %s" % database.name)
         self.__query("CREATE DATABASE %s" % database.name)
 
-    def create_user(self, credential, roles=["ALL PRIVILEGES"]):
+    def create_user(self, credential):
```
Comment (otherpirate) [inline dbaas/drivers/mysqldb.py]: Please remove roles from BaseDriver class

File: dbaas/logical/static/js/credential.js [hunk 10]
```diff
@@ -160,21 +160,39 @@
         $(document).on("click.add-credential", "#add-credential", function(e) {
             $("tbody", "#table-credentials").append(
                 "<tr class='credential'><td colspan='3'>" +
-                "<input type='text' placeholder='type username' maxlength='16' name='user' value='' />" +
+                "<input type='text' placeholder='type username' maxlength='16' id='user-roles' name='user' value='' />" +
+                "<select class='span roles' id='user-privileges' required>"+
+                "<option value='Owner' selected='selected'>Owner</option>"+
+                "<option value='Read-Write' >Read-Write</option>"+
+                "<option value='Read-Only'>Read-Only</option>"+
```
Comment (otherpirate) [inline dbaas/logical/static/js/credential.js]: These options should come from ChoicesField

**Pull request merged**